app/src/main/java/com/example/recipeapp/models/Review.java

```diff
@@ -1,21 +1,62 @@
 package com.example.recipeapp.models;
 
+import android.util.Log;
+
 import com.parse.ParseClassName;
 import com.parse.ParseObject;
 
+import java.util.Date;
+
 @ParseClassName("Review")
 public class Review extends ParseObject {
     public static final String KEY_AUTHOR = "author";
     public static final String KEY_DESCRIPTION = "description";
     public static final String KEY_RATING = "rating";
     public static final String KEY_RECIPE = "recipe";
+    private static final String TAG = "Review";
+
+    public static String calculateTimeAgo(Date createdAt) {
+
+        int SECOND_MILLIS = 1000;
+        int MINUTE_MILLIS = 60 * SECOND_MILLIS;
+        int HOUR_MILLIS = 60 * MINUTE_MILLIS;
+        int DAY_MILLIS = 24 * HOUR_MILLIS;
+
+        try {
+            createdAt.getTime();
+            long time = createdAt.getTime();
+            long now = System.currentTimeMillis();
+
+            final long diff = now - time;
+            if (diff < MINUTE_MILLIS) {
+                return "just now";
+            } else if (diff < 2 * MINUTE_MILLIS) {
+                return "a minute ago";
+            } else if (diff < 50 * MINUTE_MILLIS) {
+                return diff / MINUTE_MILLIS + " m";
+            } else if (diff < 90 * MINUTE_MILLIS) {
+                return "an hour ago";
+            } else if (diff < 24 * HOUR_MILLIS) {
+                return diff / HOUR_MILLIS + " h";
+            } else if (diff < 48 * HOUR_MILLIS) {
+                return "yesterday";
+            } else {
+                return diff / DAY_MILLIS + " d";
+            }
+        } catch (Exception e) {
+            Log.e(TAG, "Error: getRelativeTimeAgo failed", e);
+            e.printStackTrace();
+        }
+
+        return "";
+    }
```

Comment:
Don't we do this somewhere else? Can we move this into like a utils class so we only have it once? Let's chat about it at our 1:1